swift/swift.go

```diff
@@ -109,6 +110,21 @@ func (c *Client) GetReader(containerName, objectName string) (io.ReadCloser, htt
 	return requestData.RespReader, requestData.RespHeaders, err
 }
 
+// GetReadHandle creates a handler to the specified object's data.
+func (c *Client) GetReadHandle(containerName, objectName string) (io.ReadSeeker, http.Header, error) {
+	requestData := goosehttp.RequestData{}
+	err := c.touchObject(&requestData, client.HEAD, containerName, objectName)
+	if err != nil {
+		return nil, nil, err
+	}
+	l, err := strconv.ParseInt(requestData.RespHeaders.Get("Content-Length"), 10, 64)
```

Comment:
It seems a shame that this has to be done as the content length will have already been parsed  once.